src/HealthChecks.EventStore.gRPC/DependencyInjection/EventStoreHealthCheckBuilderExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using HealthChecks.EventStore.gRPC;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Extension methods to configure <see cref="EventStoreHealthCheck"/>.
+/// </summary>
+public static class EventStoreHealthCheckBuilderExtensions
+{
+    private const string NAME = "eventstore";
+
+    /// <summary>
+    /// Add a health check for EventStore services.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="connectionString">The EventStore connection string to be used.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'eventstore' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddEventStore(
+        this IHealthChecksBuilder builder,
+        string connectionString,
+        string? name = default,
+        HealthStatus? failureStatus = default,
+        IEnumerable<string>? tags = default,
+        TimeSpan? timeout = default)
+    {
+        return builder.Add(new HealthCheckRegistration(
+            name ?? NAME,
+            sp => new EventStoreHealthCheck(connectionString),

```

Comment:
A common pitfall here especially for those who implements HC in the first time. `HealthCheckRegistration`'s delegate works **each** time HC endpoint called. So you create an instance of `EventStoreHealthCheck` over and over again. Moreover, `EventStoreClient` is instantiated as well over and over again and... is not disposed. OOM sooner or later.

Check other health checks to see how to deal with this. Some implement `IDisposable` and are registered as singleton, some keep static resources (dictionaries of clients).

A suggest reviewing `NatsHealthCheck/NatsHealthCheckBuilderExtensions` and `SolrHealthCheck/SolrHealthCheckBuilderExtensions`.

Reply:
I've changed it to be a singleton, and made it implement `IDisposable` - Thanks for the help!